Paraswap-Agents/Role-Changes/src/agent.spec.ts

```diff
@@ -0,0 +1,177 @@
+import { formatBytes32String, Interface } from "ethers/lib/utils";
+import { FindingType, FindingSeverity, Finding, HandleTransaction, TransactionEvent } from "forta-agent";
+import { createAddress, TestTransactionEvent } from "forta-agent-tools/lib/tests";
+import { provideHandleTransaction } from "./agent";
+import { EVENTS_ABI } from "./utils";
+
+const TEST_SWAPPER = createAddress("0x1");
+const TEST_ACCESS_IFACE = new Interface(EVENTS_ABI);
+const IRRELEVANT_EVENT_IFACE = new Interface([
+  "event IrrelevantEvent(bytes32 indexed role, address indexed account, address indexed sender)",
+]);
+
+console.log(formatBytes32String("ROUTER_ROLE"));
+
+const createFinding = (logName: string, args: string[]) => {
+  let description = "";
+  let alertId = "";
+  let metadata = {};
+  switch (logName) {
+    case "RoleAdminChanged":
+      description = "Admin role change detected on AugustusSwapper contract";
+      alertId = "PARASWAP-2-1";
+      metadata = {
+        role: args[1],
+        previousAdminRole: args[2],
+        newAdminRole: args[3],
+      };
+      break;
+    case "RoleGranted":
+      description = "Role grant detected on AugustusSwapper contract";
+      alertId = "PARASWAP-2-2";
+      metadata = {
+        role: args[1],
+        account: args[2].slice(0, 2) + args[2].slice(26),
+        sender: args[3].slice(0, 2) + args[3].slice(26),
```

Comment:
Yeah, I agree with @e1Ru1o. Decoding the event log is actually really simple since we can use the `ethers.utils.Interface` object that was used to encode the event. For example:
```TS
const iface = new ethers.utils.Interface(["event EventName(address indexed addr)"]);

const encodedLog = iface.encodeEventLog(iface.getEvent("EventName"), [
  createAddress("0x1"), // addr
]);

const parsedLog = TEST_ACCESS_IFACE.parseLog(encodedLog);

console.log(parsedLog.args.addr)
```
By parsing the log inside the function or just sending the `parsedLog.args` to it as a parameter, there's no need to manually decode the addresses.

Reply:
Actually, as I took a look at the `utils.ts` file, I realized that the `createFinding` function there is basically the testing `createFinding`, but instead of having `logName` and `args` as parameters, it has a `LogDescription` object, which is what you get through `parseLog`.

In this case, I think adapting the `createFinding` function to receive the parsed event log as a parameter isn't even needed since it could just be imported from `utils.ts`. Then, entries like `createFinding("RoleAdminChanged", log1.topics)` could be changed to `createFinding(TEST_ACCESS_IFACE.parseLog(log1))`. How do you feel about this, @e1Ru1o and @0xtaf ?

Reply:
thank you @e1Ru1o @Hyodar for the great suggestions. I can do it the both ways, either with Interface's parseLog or decodeParameter. which one would you prefer here @e1Ru1o ?

Reply:
The equivalent of `decodeParameter` would be `abiCoder.decode` in ethers.js, right?